lib/puppet/provider/group/gpasswd.rb

```diff
@@ -0,0 +1,75 @@
+require 'puppet/provider/group/groupadd'
+
+Puppet::Type.type(:group).provide :gpasswd, :parent => Puppet::Type::Group::ProviderGroupadd do
+  require 'set'
+  require 'shellwords'
+
+  desc <<-EOM
+    Group management via `gpasswd`. This allows for local group
+    management when the users exist in a remote system.
+  EOM
+
+  commands  :addmember => 'gpasswd',
+            :delmember => 'gpasswd'
+
+  has_feature :manages_members unless %w{HP-UX Solaris}.include? Facter.value(:operatingsystem)
+
+  def addcmd
+    cmd = []
+    cmd << super.shelljoin
+
+    if @resource[:members]
+      cmd << @resource[:members].map{ |x|
+        [ command(:addmember),'-a',x,@resource[:name] ].shelljoin
+      }
+    end
+
+    # NOTE: This will create the group but may not add some members to
+    # your group if they do not exist on the system. Any member that
+    # can be added will be added in an attempt to do "the right
+    # thing".
+    cmd.join(' && ')
+  end
+
+  def members
+    retval = @objectinfo.mem
+
+    if @resource[:members] &&
+       (@resource[:attribute_membership] == :minimum) &&
```

Comment:
Unfortunately the group type has two different parameters for controlling whether the members are a partial or complete set. The `attribute_membership` is only used for generic AIX attributes. I think you want to use `auth_membership` instead.